server/src/main/java/org/opensearch/telemetry/tracing/handler/TraceableTransportResponseHandler.java

```diff
@@ -69,19 +70,23 @@ public T read(StreamInput in) throws IOException {
     @Override
     public void handleResponse(T response) {
         try (SpanScope scope = tracer.withSpanInScope(span)) {
-            delegate.handleResponse(response);
-        } finally {
+            if (response.sampled()) {
+                span.addAttribute(TracerContextStorage.SAMPLED, true);
+            }
             span.endSpan();
+        } finally {
+            delegate.handleResponse(response);
         }
     }
 
     @Override
     public void handleException(TransportException exp) {
         try (SpanScope scope = tracer.withSpanInScope(span)) {
-            delegate.handleException(exp);
-        } finally {
+            span.addAttribute(TracerContextStorage.SAMPLED, true);
```

Comment:
Little disconnected I believe. Do we guarantee we some add this attribute then it will be sampled?

Reply:
Not just with this but if we have inferred sampling enabled then this will be sampled